From 7c40eb9313dc5f2fd0500e1f6c1342c15140f52a Mon Sep 17 00:00:00 2001
From: Chelsio Cudbg <chelsiocudbg@users.noreply.github.com>
Date: Wed, 18 Feb 2026 10:23:01 -0500
Subject: [PATCH] msix changes
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

- fix the MSI-X vector calculation for ofldqsets

debugfs related changes

- show 'PM RX Cache Stats' in pm_stats debugfs file
- remove TOE-TX–related output from sge_qinfo
---
 .../ethernet/chelsio/cxgb4/cxgb4_debugfs.c    | 57 +------------------
 .../net/ethernet/chelsio/cxgb4/cxgb4_main.c   |  6 +-
 2 files changed, 3 insertions(+), 60 deletions(-)

diff --git a/drivers/net/ethernet/chelsio/cxgb4/cxgb4_debugfs.c b/drivers/net/ethernet/chelsio/cxgb4/cxgb4_debugfs.c
index 207cecfbc90c..6fd675f7090a 100644
--- a/drivers/net/ethernet/chelsio/cxgb4/cxgb4_debugfs.c
+++ b/drivers/net/ethernet/chelsio/cxgb4/cxgb4_debugfs.c
@@ -952,7 +952,7 @@ static int pm_stats_show(struct seq_file *seq, void *v)
 			   rx_cnt[i], rx_cyc[i]);
 	}
 
-	else if (CHELSIO_CHIP_VERSION(adap->params.chip) >= CHELSIO_T7) {
+	if (CHELSIO_CHIP_VERSION(adap->params.chip) >= CHELSIO_T7) {
 		u32 stats[T7_PM_RX_CACHE_NSTATS];
 
 		t4_pmrx_cache_get_stats(adap, stats);
@@ -3213,59 +3213,6 @@ static int cxgb4_sge_qinfo_uld_txq_num(struct adapter *adap,
 	return ntx;
 }
 
-static int cxgb4_sge_qinfo_uld_toe(struct seq_file *seq, int *row)
-{
-	struct t4_linux_debugfs_data *d = seq->private;
-	int r, nentries, nsharetx, nsendtx;
-	struct adapter *adap = d->adap;
-	struct sge *s = &adap->sge;
-
-	if (!cxgb4_uld_supported(adap, CXGB4_ULD_TYPE_TOE))
-		return -EOPNOTSUPP;
-
-	nsharetx = cxgb4_sge_qinfo_uld_txq_num(adap, CXGB4_ULD_TXQ_TYPE_SHARED,
-					       CXGB4_ULD_TYPE_TOE);
-	nsendtx = cxgb4_sge_qinfo_uld_txq_num(adap, CXGB4_ULD_TXQ_TYPE_SENDPATH,
-					      CXGB4_ULD_TYPE_TOE);
-	nentries = DIV_ROUND_UP(s->ofldqsets, SGE_QINFO_NUM_PER_ROW) +
-		   DIV_ROUND_UP(nsharetx, SGE_QINFO_NUM_PER_ROW) +
-		   DIV_ROUND_UP(nsendtx, SGE_QINFO_NUM_PER_ROW);
-#ifdef CONFIG_T4_MA_FAILOVER
-	nentries += DIV_ROUND_UP(s->nfailoverq, SGE_QINFO_NUM_PER_ROW);
-#endif /* CONFIG_T4_MA_FAILOVER */
-
-	if (!row)
-		return nentries;
-	r = *row;
-	if (r >= nentries) {
-		*row -= nentries;
-		return -EINVAL;
-	}
-
-	r -= DIV_ROUND_UP(s->ofldqsets, SGE_QINFO_NUM_PER_ROW);
-#ifdef CONFIG_T4_MA_FAILOVER
-	if (r < DIV_ROUND_UP(s->nfailoverq, SGE_QINFO_NUM_PER_ROW)) {
-		rx = &s->failoverq;
-		cxgb4_sge_qinfo_uld_rx(seq, r, s->nfailoverq, rx,
-				       "MA-FAILOVER");
-		return 0;
-	}
-
-	r -= DIV_ROUND_UP(s->nfailoverq, SGE_QINFO_NUM_PER_ROW);
-#endif /* CONFIG_T4_MA_FAILOVER */
-
-	if (r < DIV_ROUND_UP(nsharetx, SGE_QINFO_NUM_PER_ROW)) {
-		cxgb4_sge_qinfo_uld_tx(seq, r, CXGB4_ULD_TXQ_TYPE_SHARED,
-				       CXGB4_ULD_TYPE_TOE, "TOE-TX");
-		return 0;
-	}
-
-	r -= DIV_ROUND_UP(nsharetx, SGE_QINFO_NUM_PER_ROW);
-	cxgb4_sge_qinfo_uld_tx(seq, r, CXGB4_ULD_TXQ_TYPE_SENDPATH,
-			       CXGB4_ULD_TYPE_TOE, "TOE-SENDTX");
-	return 0;
-}
-
 static int cxgb4_sge_qinfo_uld_rdma(struct seq_file *seq, int *row)
 {
 	struct t4_linux_debugfs_data *d = seq->private;
@@ -3716,7 +3663,6 @@ static int sge_qinfo_show(struct seq_file *seq, void *v)
 	SGE_QINFO_CALL(eth_trace);
 	SGE_QINFO_CALL(eth_mirror);
 	SGE_QINFO_CALL(eth_vxlan);
-	SGE_QINFO_CALL(uld_toe);
 	SGE_QINFO_CALL(uld_rdma);
 	SGE_QINFO_CALL(uld_iscsi);
 	SGE_QINFO_CALL(uld_iscsit);
@@ -3758,7 +3704,6 @@ static int sge_queue_entries(struct seq_file *seq)
 	SGE_QINFO_NUM_CALL(eth_trace);
 	SGE_QINFO_NUM_CALL(eth_mirror);
 	SGE_QINFO_NUM_CALL(eth_vxlan);
-	SGE_QINFO_NUM_CALL(uld_toe);
 	SGE_QINFO_NUM_CALL(uld_rdma);
 	SGE_QINFO_NUM_CALL(uld_iscsi);
 	SGE_QINFO_NUM_CALL(uld_iscsit);
diff --git a/drivers/net/ethernet/chelsio/cxgb4/cxgb4_main.c b/drivers/net/ethernet/chelsio/cxgb4/cxgb4_main.c
index e101650cf44f..1d743abe3c10 100644
--- a/drivers/net/ethernet/chelsio/cxgb4/cxgb4_main.c
+++ b/drivers/net/ethernet/chelsio/cxgb4/cxgb4_main.c
@@ -5350,7 +5350,6 @@ static int cxgb4_enable_irqs(struct adapter *adap)
 {
 	u32 eth_need, uld_need = 0, ethofld_need = 0, mirror_need = 0;
 	u32 ethqsets = 0, ofldqsets = 0, eoqsets = 0, mirrorqsets = 0;
-	u8 num_uld = 0;
 	u8 nchan = adap->params.nports;
 	u32 i, want, need, num_vec, flags = 0;
 	struct sge *s = &adap->sge;
@@ -5369,9 +5368,8 @@ static int cxgb4_enable_irqs(struct adapter *adap)
 #endif
 	eth_need = need;
 	if (is_uld(adap)) {
-		num_uld = adap->num_ofld_uld + adap->num_uld;
-		want += num_uld * s->ofldqsets;
-		uld_need = num_uld * nchan;
+		want += s->ofldqsets;
+		uld_need = adap->params.num_up_cores * nchan;
 		need += uld_need;
 	}
 
-- 
2.39.1

